### PR TITLE
workflows: conditional halt in send_and_wait_robotupload

### LINKED ISSUE #2008: authors: workflow for updates should not wait for robotupload
The following commit https://github.com/inspirehep/inspire-next/commit/c97fb948a423a3ac9f09b50010863b19d85d1a13 changed the author workflow in a way that will make the workflow never complete.

The workflow needs to do:

* On update: send robotupload (in HoldingPen mode) and create a ticket.

* On new: send robotupload (in insert mode), create a ticket and wait. Once callback is received resume workflow.

So reusing `send_and_wait_robotupload` for both cases might not be possible (without modifying that function)

### PR DESCRIPTION
**Pull request by jmartinm**
* Renames send_and_wait_robotupload to send_robotupload.

* Halts workflow only when a callback URL is set. (closes #2008)

### CODE REVIEW

File: inspirehep/modules/workflows/tasks/submission.py [hunk 1]
```diff
@@ -237,19 +237,21 @@ def send_and_wait_robotupload(
 ):
     """Get the MARCXML from the model and ship it."""
```
Comment (david-caro) [inline inspirehep/modules/workflows/tasks/submission.py]: Do you mind adding here a note about the callback_url parameter?

File: inspirehep/modules/workflows/tasks/submission.py [hunk 2]
```diff
@@ -237,19 +237,21 @@ def send_and_wait_robotupload(
...
         from inspirehep.utils.robotupload import make_robotupload_marcxml
 
-        combined_callback_url = os.path.join(
-            current_app.config["SERVER_NAME"],
-            callback_url
-        )
-        if not combined_callback_url.startswith('http'):
-            combined_callback_url = "https://{0}".format(
-                combined_callback_url
+        combined_callback_url = ""
```
Comment (david-caro) [inline inspirehep/modules/workflows/tasks/submission.py]: I'm not sure how legacy interprets the callback_url parameter and such... maybe @kaplun can shed some light? (I tried for 5 min to find out where is the code of the batchupload on the old code... no luck).

Comment (david-caro) [inline inspirehep/modules/workflows/tasks/submission.py]: Another option is just try it out on qa

Comment (jmartinm) [inline inspirehep/modules/workflows/tasks/submission.py]: I tried yesterday with my localhost connected to inspireheptest and passing empty meant that no parameter arrived to the `robotupload` command so should be fine.

Comment (david-caro) [inline inspirehep/modules/workflows/tasks/submission.py]: 👍

**Comment by jmartinm** [comment]
Updated.

**Pull request merged**